src/Database/DriverInterface.php

```diff
@@ -0,0 +1,260 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         3.6.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Database;
+
+use Cake\Database\Query;
+use Cake\Database\Statement\PDOStatement;
+use InvalidArgumentException;
+use PDO;
+use PDOException;
+
+/**
+ * Interface for database driver.
+ */
+interface DriverInterface
+{
+    /**
+     * Establishes a connection to the database server.
+     *
+     * @return bool True on success, false on failure.
+     */
+    public function connect();
+
+    /**
+     * Disconnects from database server.
+     *
+     * @return void
+     */
+    public function disconnect();
+
+    /**
+     * Returns correct connection resource or object that is internally used.
+     *
+     * @return mixed Connection object used internally.
```

Comment:
Can we be more specific, e.g. object or some interface?

Reply:
No we can't, since the whole premise of this interface is the ability to use a non PDO connection, which at this point could be anything :)

Reply:
not int^^ at least object I believe.